alacritty/src/input/keyboard.rs

```diff
@@ -107,15 +107,22 @@ impl<T: EventListener, A: ActionContext<T>> Processor<T, A> {
         mods: ModifiersState,
     ) -> bool {
         if mode.contains(TermMode::REPORT_ALL_KEYS_AS_ESC) {
-            true
-        } else if mode.contains(TermMode::DISAMBIGUATE_ESC_CODES) {
+            return true;
+        }
+
+        let disambiguate = if mode.contains(TermMode::DISAMBIGUATE_ESC_CODES) {
             let on_numpad = key.location == KeyLocation::Numpad;
             let is_escape = key.logical_key == Key::Named(NamedKey::Escape);
             is_escape || (!mods.is_empty() && mods != ModifiersState::SHIFT) || on_numpad
         } else {
```

Comment:
The else branch is just `false`, so this can probably be written as a single conditional?

You could still keep the variables extracted for documentation purposes, but they're also pretty self-documenting.